homeassistant/components/zwave_js/update.py

```diff
@@ -93,6 +98,36 @@ def _update_on_status_change(self, _: dict[str, Any]) -> None:
         self._status_unsub = None
         self.hass.async_create_task(self._async_update())
 
+    @callback
+    def _update_progress(self, evt: dict[str, Any]) -> None:
+        """Update install progress on event."""
+        progress: FirmwareUpdateProgress = evt["firmware_update_progress"]
+        if not self._latest_version_firmware:
+            return
+        # We will assume that each file in the firmware update represents an equal
+        # percentage of the overall progress. This is likely not true because each file
+        # may be a different size, but it's the best we can do since we don't know the
+        # total number of fragments across all files.
+        self._attr_in_progress = floor(
+            100
+            * (
+                self._num_files_installed
+                + (progress.sent_fragments / progress.total_fragments)
+            )
+            / len(self._latest_version_firmware.files)
```

Comment:
I don't understand this formula. Why do we add the number of installed files to the progress quota?

Reply:
the idea is that if there are multiple files, we need a way to track progress to 100% across all of the files, but we don't know the total number of fragments in advance and the callback is separate from the place where we track when a file is done installing (the logic is in the service function). So if there are two files, one file is done, and the next file is 50% complete, this formula would work out to 150/2 = 75. The first file is 50%, the second file is 25% because it is halfway done. Does that make sense?

Reply:
Yes, with your explanation it makes sense. Thanks!